[BUG] No Drops When Breaking Blocks On Island

**Describe the bug**
When I try to break any blocks on the island nothing drops so it is impossible to complete the first mission without using /completemissionset GETTING_STARTED. 

**To Reproduce**
Steps to reproduce the behavior:
1. Go on versions '1.8.9-1.21'
2. Join server
3. Try to break a block, for example dirt, grass, or wood.
4. See error

**Expected behavior**
The ability to get the block you broke.

**Screenshots**
![Screenshot 2024-06-23 083517](https://github.com/Swofty-Developments/HypixelSkyBlock/assets/104693998/153484d4-edd0-4e31-89e7-9722186dd30e)

When I try to break the block I get this error below, I attached the full log:

PS C:\Users\aiden\OneDrive\Desktop\SB Server\game_server> java --enable-preview -jar SkyBlockCore.jar ISLAND
2024-06-23 08:54:44 [main] net.swofty.loader.SkyBlock.lambda$main$1() INFO: Found TypeLoader: THE_FARMING_ISLANDS
2024-06-23 08:54:44 [main] net.swofty.loader.SkyBlock.lambda$main$1() INFO: Found TypeLoader: ISLAND
2024-06-23 08:54:46 [main] net.swofty.types.generic.SkyBlockGenericLoader.initialize() ERROR: Not all songs have been found, song service will not start for this instance.
2024-06-23 08:54:47 [main] net.swofty.type.island.TypeIslandLoader.onInitialize() INFO: TypeIslandLoader initialized!
2024-06-23 08:54:47 [main] net.swofty.loader.SkyBlock.main() INFO: Initializing proxy support...
2024-06-23 08:54:47 [Thread-2] net.minestom.server.ServerProcessImpl.start() INFO: Starting SkyBlock server.
2024-06-23 08:54:48 [Thread-2] net.minestom.server.ServerProcessImpl.start() INFO: SkyBlock server started successfully.
2024-06-23 08:54:48 [Thread-2] net.swofty.loader.SkyBlock.lambda$main$4() INFO: Started server on port 20000 in 3401ms
2024-06-23 08:54:48 [Thread-2] net.swofty.loader.SkyBlock.lambda$main$4() INFO: Server Type: ISLAND
2024-06-23 08:54:48 [Thread-2] net.swofty.loader.SkyBlock.lambda$main$4() INFO: Internal ID: 24595ae0-2619-46bc-837a-807e53aa7dc6
2024-06-23 08:55:00 [Ms-worker-1] net.swofty.types.generic.SkyBlockGenericLoader.lambda$initialize$30() INFO: Received new player: JustAiden (f539d1d0-d861-45cf-8c16-a6400e6b84e1)
2024-06-23 08:55:00 [ForkJoinPool.commonPool-worker-14] net.swofty.types.generic.event.actions.player.data.ActionPlayerDataLoad.run() INFO: New player joined: JustAiden...
2024-06-23 08:55:01 [] net.swofty.types.generic.event.actions.player.data.ActionPlayerDataSpawn.run() INFO: Loading player data for JustAiden...
2024-06-23 08:55:11 [Ms-TickScheduler] net.swofty.types.generic.event.SkyBlockEventHandler.lambda$register$3() INFO: Exception occurred while running event Method with event type PlayerBlockBreakEvent
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:115)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at net.swofty.types.generic.event.SkyBlockEventHandler.runEvent(SkyBlockEventHandler.java:156)
        at net.swofty.types.generic.event.SkyBlockEventHandler.lambda$register$3(SkyBlockEventHandler.java:106)
        at net.minestom.server.event.EventListener$Builder$1.run(EventListener.java:151)
        at net.minestom.server.event.EventNodeImpl$Handle.callListener(EventNodeImpl.java:491)
        at net.minestom.server.event.EventNodeImpl$Handle.lambda$listenersConsumer$9(EventNodeImpl.java:429)
        at net.minestom.server.event.EventNodeImpl$Handle.lambda$createConsumer$5(EventNodeImpl.java:394)
        at net.minestom.server.event.EventNodeImpl$Handle.lambda$createConsumer$5(EventNodeImpl.java:402)
        at net.minestom.server.event.EventNodeImpl$Handle.call(EventNodeImpl.java:327)
        at net.minestom.server.event.EventNode.call(EventNode.java:190)
        at net.minestom.server.event.EventDispatcher.call(EventDispatcher.java:10)
        at net.minestom.server.instance.InstanceContainer.breakBlock(InstanceContainer.java:234)
        at net.minestom.server.instance.SharedInstance.breakBlock(SharedInstance.java:41)
        at net.minestom.server.instance.Instance.breakBlock(Instance.java:202)
        at net.minestom.server.listener.PlayerDiggingListener.breakBlock(PlayerDiggingListener.java:177)
        at net.minestom.server.listener.PlayerDiggingListener.finishDigging(PlayerDiggingListener.java:112)
        at net.minestom.server.listener.PlayerDiggingListener.playerDiggingListener(PlayerDiggingListener.java:46)
        at net.minestom.server.listener.manager.PacketListenerManager.lambda$setPlayListener$2(PacketListenerManager.java:163)
        at net.minestom.server.listener.manager.PacketListenerManager.processClientPacket(PacketListenerManager.java:132)
        at net.minestom.server.entity.Player.lambda$interpretPacketQueue$12(Player.java:2128)
        at org.jctools.queues.MpscUnboundedXaddArrayQueue.drain(MpscUnboundedXaddArrayQueue.java:312)
        at net.minestom.server.entity.Player.interpretPacketQueue(Player.java:2128)
        at java.base/java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:891)
        at java.base/java.util.concurrent.CopyOnWriteArraySet.forEach(CopyOnWriteArraySet.java:425)
        at net.minestom.server.network.ConnectionManager.tick(ConnectionManager.java:368)
        at net.minestom.server.ServerProcessImpl$TickerImpl.tick(ServerProcessImpl.java:375)
        at net.minestom.server.thread.TickSchedulerThread.run(TickSchedulerThread.java:29)
Caused by: java.lang.NullPointerException: Cannot invoke "net.swofty.types.generic.region.mining.MineableBlock.getFortuneType()" because "mineableBlock" is null
        at net.swofty.types.generic.event.actions.player.region.ActionRegionBlockBreak.run(ActionRegionBlockBreak.java:73)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        ... 27 more

--- a/type.generic/src/main/java/net/swofty/types/generic/event/actions/player/region/ActionRegionBlockBreak.java
+++ b/type.generic/src/main/java/net/swofty/types/generic/event/actions/player/region/ActionRegionBlockBreak.java
@@ -1,10 +1,12 @@
 package net.swofty.types.generic.event.actions.player.region;
 
+import net.kyori.adventure.text.Component;
 import net.minestom.server.coordinate.Pos;
 import net.minestom.server.event.player.PlayerBlockBreakEvent;
 import net.minestom.server.instance.SharedInstance;
 import net.minestom.server.instance.block.Block;
 import net.minestom.server.item.Material;
+import net.swofty.commons.StringUtility;
 import net.swofty.types.generic.SkyBlockConst;
 import net.swofty.types.generic.entity.DroppedItemEntityImpl;
 import net.swofty.types.generic.event.EventNodes;
@@ -19,7 +21,6 @@ import net.swofty.types.generic.region.mining.MineableBlock;
 import net.swofty.types.generic.user.SkyBlockPlayer;
 import net.swofty.types.generic.event.SkyBlockEvent;
 import net.swofty.types.generic.event.custom.CustomBlockBreakEvent;
-import net.swofty.types.generic.utility.MaterialQuantifiableRandom;
 
 public class ActionRegionBlockBreak implements SkyBlockEventClass {
 
@@ -69,15 +70,22 @@ public class ActionRegionBlockBreak implements SkyBlockEventClass {
                     player, item.getMaterial(), event.getBlockPosition()
             ));
 
-            MineableBlock mineableBlock = MineableBlock.get(block);
-            Double fortune = player.getStatistics().allStatistics().getOverall(mineableBlock.getFortuneType());
-            Double dropMultiplicator = (1 + (fortune*0.01));
-            Integer drops = mineableBlock.getDrops().getAmount(dropMultiplicator);
+            MineableBlock mineableBlock;
+            Integer drops;
+            
+            try {
+                mineableBlock = MineableBlock.get(block);
+                Double fortune = player.getStatistics().allStatistics().getOverall(mineableBlock.getFortuneType());
+                Double dropMultiplicator = (1 + (fortune*0.01));
+                drops = mineableBlock.getDrops().getAmount(dropMultiplicator);
+            } catch (NullPointerException e) {
+                drops = 1;
+            }
 
             /**
              * Handle block dropping
              */
-            SkyBlockItem skyBlockItem = new SkyBlockItem(item.getAttributeHandler().getPotentialType(), drops);
+            SkyBlockItem skyBlockItem = new SkyBlockItem(item.getItemStackBuilder().amount(drops).build());
             if (player.getSkyBlockExperience().getLevel().asInt() >= 6) {
                 player.addAndUpdateItem(skyBlockItem);
             } else {
@@ -85,14 +93,6 @@ public class ActionRegionBlockBreak implements SkyBlockEventClass {
                 Pos pos = Pos.fromPoint(event.getBlockPosition());
                 // Move the dropped item to the center of the block
                 pos = pos.add(0.5, 0.5, 0.5);
-                // Move block closer to player by 0.5 blocks
-                Pos playerPos = player.getPosition().add(0, 0.5, 0);
-                double x = playerPos.x() - pos.x();
-                double y = playerPos.y() - pos.y();
-                double z = playerPos.z() - pos.z();
-                double distance = Math.sqrt(x * x + y * y + z * z);
-                double multiplier = 1.4 / distance;
-                pos = pos.add(x * multiplier, y * multiplier * 3, z * multiplier);
 
                 droppedItem.setInstance(player.getInstance(), pos);
                 droppedItem.addViewer(player);

--- a/type.generic/src/main/java/net/swofty/types/generic/item/items/weapon/EndSword.java
+++ b/type.generic/src/main/java/net/swofty/types/generic/item/items/weapon/EndSword.java
@@ -46,8 +46,8 @@ public class EndSword extends SkyBlockValueEvent implements CustomSkyBlockItem, 
         PlayerDamageMobValueUpdateEvent event = (PlayerDamageMobValueUpdateEvent) tempEvent;
         SkyBlockPlayer player = event.getPlayer();
         SkyBlockItem item = new SkyBlockItem(player.getItemInMainHand());
-        if(item.isNA() || item.isAir()) return;
-        if(item.getAttributeHandler().getPotentialClassLinker() != ItemTypeLinker.END_SWORD) return;
+        if (item.isNA() || item.isAir()) return;
+        if (item.getAttributeHandler().getPotentialClassLinker() != ItemTypeLinker.END_SWORD) return;
         if (event.getMob().getEntityType() == EntityType.ENDER_DRAGON || event.getMob().getEntityType() == EntityType.ENDERMITE || event.getMob().getEntityType() == EntityType.ENDERMAN) {
             event.setValue((((float) event.getValue()) * 2));
         }

--- a/type.generic/src/main/java/net/swofty/types/generic/item/items/weapon/SpiderSword.java
+++ b/type.generic/src/main/java/net/swofty/types/generic/item/items/weapon/SpiderSword.java
@@ -48,8 +48,8 @@ public class SpiderSword extends SkyBlockValueEvent implements CustomSkyBlockIte
         PlayerDamageMobValueUpdateEvent event = (PlayerDamageMobValueUpdateEvent) tempEvent;
         SkyBlockPlayer player = event.getPlayer();
         SkyBlockItem item = new SkyBlockItem(player.getItemInMainHand());
-        if(item.isNA() || item.isAir()) return;
-        if(item.getAttributeHandler().getPotentialClassLinker() != ItemTypeLinker.SPIDER_SWORD) return;
+        if (item.isNA() || item.isAir()) return;
+        if (item.getAttributeHandler().getPotentialClassLinker() != ItemTypeLinker.SPIDER_SWORD) return;
         if (event.getMob().getEntityType() == EntityType.SPIDER || event.getMob().getEntityType() == EntityType.CAVE_SPIDER || event.getMob().getEntityType() == EntityType.SILVERFISH) {
             event.setValue((((float) event.getValue()) *2));
         }

--- a/type.generic/src/main/java/net/swofty/types/generic/item/items/weapon/UndeadSword.java
+++ b/type.generic/src/main/java/net/swofty/types/generic/item/items/weapon/UndeadSword.java
@@ -47,8 +47,8 @@ public class UndeadSword extends SkyBlockValueEvent implements CustomSkyBlockIte
         PlayerDamageMobValueUpdateEvent event = (PlayerDamageMobValueUpdateEvent) tempEvent;
         SkyBlockPlayer player = event.getPlayer();
         SkyBlockItem item = new SkyBlockItem(player.getItemInMainHand());
-        if(item.isNA() || item.isAir()) return;
-        if(item.getAttributeHandler().getPotentialClassLinker() != ItemTypeLinker.UNDEAD_SWORD) return;
+        if (item.isNA() || item.isAir()) return;
+        if (item.getAttributeHandler().getPotentialClassLinker() != ItemTypeLinker.UNDEAD_SWORD) return;
         if (event.getMob().getEntityType() == EntityType.ZOMBIE || event.getMob().getEntityType() == EntityType.ZOMBIFIED_PIGLIN || event.getMob().getEntityType() == EntityType.SKELETON || event.getMob().getEntityType() == EntityType.WITHER_SKELETON) {
             event.setValue((((float) event.getValue()) *2));
         }

--- a/type.generic/src/main/java/net/swofty/types/generic/item/items/weapon/WitherBow.java
+++ b/type.generic/src/main/java/net/swofty/types/generic/item/items/weapon/WitherBow.java
@@ -49,9 +49,9 @@ public class WitherBow extends SkyBlockValueEvent implements CustomSkyBlockItem,
         PlayerDamageMobValueUpdateEvent event = (PlayerDamageMobValueUpdateEvent) tempEvent;
         SkyBlockPlayer player = event.getPlayer();
         SkyBlockItem item = new SkyBlockItem(player.getItemInMainHand());
-        if(item.isNA() || item.isAir()) return;
-        if(item.getAttributeHandler().getPotentialClassLinker() != ItemTypeLinker.WITHER_BOW) return;
-        if ( event.getMob().getEntityType() == EntityType.WITHER_SKELETON || event.getMob().getEntityType() == EntityType.WITHER) {
+        if (item.isNA() || item.isAir()) return;
+        if (item.getAttributeHandler().getPotentialClassLinker() != ItemTypeLinker.WITHER_BOW) return;
+        if (event.getMob().getEntityType() == EntityType.WITHER_SKELETON || event.getMob().getEntityType() == EntityType.WITHER) {
             event.setValue((((float) event.getValue()) *2));
         }
     }

